packages/beacon-node/test/e2e/doppelganger/doppelganger.test.ts

```diff
@@ -171,30 +156,16 @@ describe.skip("doppelganger / doppelganger test", function () {
     });
     afterEachCallbacks.push(() => Promise.all(validator0WithoutDoppelganger.map((v) => v.close())));
 
-    expect(validator0WithDoppelganger[0].isRunning).to.be.equal(
-      true,
-      "validator with doppelganger protection should be running"
-    );
-    expect(validator0WithoutDoppelganger[0].isRunning).to.be.equal(
-      true,
-      "validator without doppelganger protection should be running before first epoch"
-    );
```

Comment:
The amount of information we lose here in terms of reading the test code and possible errors when assertions fails is not ideal

Reply:
I agree and reason is that we didn't had the tests in ideal position earlier. 

These are the best practices; 

1. Have simple and straight forward expectations. 
2. All the information related to test should be capture from test description, so if that fails you know what's wrong. 
3. If test failure needs to have further explanation, that implies that test description need to be more informative. 
4. If a test need mixed or confused expectations, it needs to be divided into multiple test cases. 
5. If an assertion is complex then should be extracted as custom assertion so it have consistent messaging across the tests. 

These practices can't be enforce if assertion libraries are flexible like chai. So why Jest, Vitest and similar libraries don't provide feature for custom messages. 

Let's take an example. 

We need to compare `a` is greater than `b` and if we do with nodejs assertion module, we will have. 

```js
assert(a > b); 
```
Such assertion is confusing unless provided more details as; 

```js
assert(a > b, "a is not greater than b"); 
```

The problem with this approach is that one developer may write a different assertion message and other may write different. 

But if you use frameworks like Jest, Vitest they provide special assertions e.g.

```ks
expect(a). toBeGreaterThan(b); 
```
With this assertion it's very clear and will have consistent error messages across every project. 

---

Now take our example code. 

```js
 expect(validator0WithDoppelganger[0].isRunning).to.be.equal(
      true,
      "validator with doppelganger protection should be running"
    );
```

Having such assertion messages clearly sign of bad practice. It's a a clear an test case, not an assertion message. It should have been like this;

```js
it("validator with doppelganger protection should be running", () => {
    expect(validator0WithDoppelganger[0].isRunning).to.be.true;
})
```

If that test was failed, you clearly knows what's wrong and you don't need custom assertions messages to explain.

Reply:
> The problem with this approach is that one developer may write a different assertion message and other may write different.

I don't mind different message as long as those are correct and informative. In my opinion the assertion itself only tells you what is expected but not why. Could split out each assertion into its own test case but would have to see how practical that is.

Definitely have to update the [tests-style-guide](https://github.com/ChainSafe/lodestar/blob/unstable/CONTRIBUTING.md#tests-style-guide) as it specifically mentions assertion messages.

Reply:
Should at least add previous messages as comments, as it was done in other places

Reply:
Yes we need to update that, never saw that earlier.

Reply:
> Now take our example code.

Just to clarify this example, does it mean that what is currently in the assertion message should be moved up to `it` message and what is in the `it` message should be moved up to `describe`, and then possibly have nested describe blocks?

Reply:
Logically yes.